### DISCUSSION TITLE
Can't embed more than 2048 items

### DISCUSSION THREAD

**Discussion by ahmedkooli** [michaelfeil/infinity#447] [Q&A] [answered]
Hey, I get this json as output when I try to embed a list of more than 2048 items:
```
{'type': 'too_long',
  'loc': ['body', 'text', 'input', 'list[constrained-str]'],
  'msg': 'List should have at most 2048 items after validation, not 2049'}
```
I have a few questions regarding this:
1- Where is this limitation coming from?
2- Can this limit be changed?
3- Why doesn't the batch size take care of such large inputs?

Thanks a lot for your help!

**Comment by michaelfeil** [comment, marked as answer]
@ahmedkooli There are multiple issues explaining why. Its mostly about good practices & maximum payload size / timeout etc of common cloud infrastructure.

#307 #338
